src/engine/srv.c

```diff
@@ -602,27 +609,38 @@ dss_start_one_xstream(hwloc_cpuset_t cpus, int xs_id)
 	 * as it is only for EC/checksum/compress offloading.
 	 */
 	if (dss_helper_pool) {
-		comm = (xs_id == 0) || (xs_id >= dss_sys_xs_nr &&
-				xs_id < (dss_sys_xs_nr + 2 * dss_tgt_nr));
+		comm =  (xs_id == 0) || /* DSS_XS_SYS */
+			(xs_id == 2) || /* DSS_XS_SWIM */
```

Comment:
I thought it again, there is two cases that -
1. no dedicate core available for SWIM, so SWIM can share to use XS0 and cart context 0 with RDB same as before (may need not create separate XS and cart context for SWIM in this case).
2. with dedicate core available for SWIM, SWIM will create separate XS and cart context. But a side-effect is that if create SWIM cart context following XS0's cart context, this means the first VOS target's main XS' cart context will have one more offset compared to case 1. This may cause some inconvenience when client sends DAOS rpc to server as there will have two different cases.

To avoid that difference, for case 2, we may create the SWIM's XS and cart context as the last cart context, to avoid affect the DAOS_TGT0_OFFSET/DAOS_IO_CTX_ID calculation for client. As SWIM is completely server-side module and need not interact with client.
Just FYI.

Reply:
@liuxuezhao I understand your point but as far as I understand it was decision about separate XS for SWIM and shifting all VOS XS by one to avoid any issue with addressing them. The only question was about how to schedule this new XS. I understand the latest suggestion was to share the same core with system XS if not enough cores and use separate core otherwise.

@johannlombardi could you clarify this one more time please here?